From 5b62793cdbf9f9a14de3cec0971f5fc688784165 Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Mon, 27 Jul 2020 10:10:10 +0200
Subject: [PATCH 6/7] arm/arm64: Expand return type of smc helpers to long

This allows to evaluate all 64 bits on arm64. Will be needed for SDEI.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 hypervisor/arch/arm-common/smccc.c      | 2 +-
 hypervisor/arch/arm/include/asm/smc.h   | 2 +-
 hypervisor/arch/arm64/include/asm/smc.h | 4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/hypervisor/arch/arm-common/smccc.c b/hypervisor/arch/arm-common/smccc.c
index c87b9fb1..bae3ad75 100644
--- a/hypervisor/arch/arm-common/smccc.c
+++ b/hypervisor/arch/arm-common/smccc.c
@@ -47,7 +47,7 @@ static void trust_dispatch(struct trap_context *ctx)
 void smccc_discover(void)
 {
 	struct per_cpu *cpu_data = this_cpu_data();
-	int ret;
+	long ret;
 
 	cpu_data->smccc_feat_workaround_1 = ARM_SMCCC_NOT_SUPPORTED;
 	cpu_data->smccc_feat_workaround_2 = ARM_SMCCC_NOT_SUPPORTED;
diff --git a/hypervisor/arch/arm/include/asm/smc.h b/hypervisor/arch/arm/include/asm/smc.h
index 0f5e857f..b70af8be 100644
--- a/hypervisor/arch/arm/include/asm/smc.h
+++ b/hypervisor/arch/arm/include/asm/smc.h
@@ -13,7 +13,7 @@
 /* for gcc < 5 */
 asm (".arch_extension sec\n");
 
-static inline int smc(unsigned long id)
+static inline long smc(unsigned long id)
 {
 	register unsigned long __id asm("r0") = id;
 
diff --git a/hypervisor/arch/arm64/include/asm/smc.h b/hypervisor/arch/arm64/include/asm/smc.h
index 9c7bc6ab..27e5397b 100644
--- a/hypervisor/arch/arm64/include/asm/smc.h
+++ b/hypervisor/arch/arm64/include/asm/smc.h
@@ -10,7 +10,7 @@
  * the COPYING file in the top-level directory.
  */
 
-static inline int smc(unsigned long id)
+static inline long smc(unsigned long id)
 {
 	register unsigned long __id asm("r0") = id;
 
@@ -21,7 +21,7 @@ static inline int smc(unsigned long id)
 	return __id;
 }
 
-static inline int smc_arg1(unsigned long id, unsigned long par1)
+static inline long smc_arg1(unsigned long id, unsigned long par1)
 {
 	register unsigned long __id asm("r0") = id;
 	register unsigned long __par1 asm("r1") = par1;
-- 
2.17.1

